### DISCUSSION TITLE
Is it possible to use local js files in Bruno?

### DISCUSSION THREAD

**Discussion by iuri-aramos** [usebruno/bruno#657] [Q&A] [answered]
I am trying to use a simple js file from the request folder and use it in the script section, but the file is not loaded.

![image](https://github.com/usebruno/bruno/assets/13989950/58494baf-c4e3-41d7-af10-79ae76bcc96e)
![image](https://github.com/usebruno/bruno/assets/13989950/bb3ff890-f51d-4016-b600-08e3fb2f4ed0)
![image](https://github.com/usebruno/bruno/assets/13989950/4f0fe68b-b958-46b9-9cc2-e91dac0718f9)

In the same way, I would like to create another script outside Bruno that can log the requests and responses, just calling the function from a js file, how the picture shows.

Is it possible to execute this strategy?

**Comment by helloanoop** [comment, marked as answer]
This is possible in Bruno :)
I think there is a typo. Your pre-request script should be more like

```javascript
const helloWorld = require("./test");

helloWorld();
``` 

You can also checkout https://github.com/usebruno/bruno/discussions/385 for some script inspirations.

**Comment by iuri-aramos** [comment]
Thank you @helloanoop, it has worked.

**Reply by helloanoop** [reply]
Happy scripting in Bruno!

**Comment by ElSeniorMikael** [comment]
Hello, im facing a problem that i can't seem to understand.

I'm trying to use a custom libraries im making to do my testing and all. I built a parent folder that hold all my custom code, inside that folder i will have all my constants (e.g:: json schema) and also have an utils folder for function that i will use, so my problem is, i made a function to validate my json schema, and im trying to export it. whenever i launch my test it fail because it couldnt find my validate function.

I manage to export string, number, object but i can't seem to be able to export an function. Any help pls ?

"ValidateJsonSchema.js"
`
const Ajv = require("ajv");
const ajv = new Ajv();

function validateJsonSchema() {
  // console.log("toto", ajv.validate(schema, data));
  // return ajv.validate(schema, data);
}

module.exports = {
  validateJsonSchema: "validateJsonSchema",
  titi: () => {},
  toto: "toto",
};
`
utils/index.js
`
const validateJsonSchema = require("./validateJsonSchema");

module.exports = {
  ...validateJsonSchema,
};
`

Libraries/index.js

`
const constants = require("./constants");
const utils = require("./utils");

module.exports = {
  constants,
  utils,
};
`
